doc/source/whatsnew/v1.0.0.rst

```diff
@@ -150,6 +150,8 @@ Other enhancements
 - Roundtripping DataFrames with nullable integer or string data types to parquet
   (:meth:`~DataFrame.to_parquet` / :func:`read_parquet`) using the `'pyarrow'` engine
   now preserve those data types with pyarrow >= 1.0.0 (:issue:`20612`).
+- :meth:`DataFrame.sort_values`, :meth:`DataFrame.sort_index`, :meth:`Series.sort_index`, and :meth:`Series.sort_index`
+   now support the ``key`` argument which allows for custom sorting orders (:issue:`3942`)
```

Comment:
add an example in the appropriate user docs section as well using key

Reply:
This will probably go in `doc/source/getting_started/basics.rst`.

Reply:
I added this to the inline docs in `core/generic.py`. I can add it to `doc/source/getting_started/basics.rst`, but it didn't seem as natural there.